activity/journal.go

```diff
@@ -37,18 +40,48 @@ func (j Journal) registerCommand() {
 	defer mux.Unlock()
 
 	// Check again in case something else beat us to it.
-	if _, ok := jidToHistory[j.ID]; ok {
+	if _, ok := storedJournals[j.ID]; ok {
 		return
 	}
 
 	// Register the command.
-	jidToHistory[j.ID] = struct{}{}
-	if j.Start.IsZero() {
-		j.Start = time.Now()
-	}
+	storedJournals[j.ID] = len(history)
 	history = append(history, j)
 }
 
+// Record writes a new entry to the journal. It creates a new file for the journal if needed, then
+// appends the message to that journal. Journals are stored in the user's cache directory under
+// `wash/activity/ID.log`.
+func (j Journal) Record(msg string, a ...interface{}) {
+	log.Debugf(msg, a...)
+
+	// This is a single-use cache, so pass in an empty category.
+	obj, err := journalFileCache.GetOrUpdate("", j.ID, expires, true, func() (interface{}, error) {
+		jdir := Dir()
+		if err := os.MkdirAll(jdir, 0750); err != nil {
+			return nil, err
+		}
+
+		lpath := filepath.Join(jdir, j.ID+".log")
```

Comment:
Nit: It'd be nice to define a `j.FilePath()` method to encapsulate the whole `j.ID + ".log"` computation since it is also used in `Open`.